<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Tade Souaiaia" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Input Data - BridgePRS</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link href="../css/extra.css" rel="stylesheet" />
        <link href="../css/style.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Input Data";
        var mkdocs_page_input_path = "guide_input.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]--> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> BridgePRS
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                </li>
              </ul>
              <p class="caption"><span class="caption-text">Requirements</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../req_system/">System</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../req_software/">Software</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../req_mac/">Mac Specific</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Quick start tutorial</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_demo/">Example Run</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../quikstart_result/">Output directories</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Guide</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../guide_background/">Background</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_commands/">Commands</a>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">Input Data</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#the-ld-reference-panel">The LD Reference Panel</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#config-files">Config Files</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#creating-a-config-file">Creating a Config File:</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#config-files-targetbase-population-data">Config Files: Target/Base Population Data:</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#file-specifications">File Specifications</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#1-sumstats-data">1) Sumstats Data</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#2-genotype-files">2) Genotype Files</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#3-phenotype-files">3) Phenotype Files</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#4-qc-snp-list">4) QC SNP List</a>
    </li>
        </ul>
    </li>
    </ul>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_output/">Output Data</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_usecases/">Use Cases</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_args/">All Arguments</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_customization/">Customization</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../guide_shell/">Using Bash</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Misc</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../misc_faq/">Frequently Asked Questions</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../misc_downloads/">Direct Downloads</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../misc_log/">Update Log</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">BridgePRS</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Guide</li>
      <li class="breadcrumb-item active">Input Data</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/tadesouaiaia/BridgePRS-website/edit/master/docs/guide_input.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <p><img alt="Screenshot" src="../img/slim/guide_logo2.png" /> </p>
<h1 id="input-data">Input Data</h1>
<h2 id="the-ld-reference-panel">The LD Reference Panel</h2>
<p>BridgePRS requires representative genotype data in PLINK binary (bed)
format to estimate LD.  A thinned LD reference panel (19734 SNPs)
for use with the <a href="../quikstart_data/">quick start tutorial</a> is included 
in the BridgePRS download (<strong>data/1000G_sample</strong>). Three full-size 1000 Genomes panels 
covering <strong>AFR</strong>, <strong>EUR</strong>, <strong>EAS</strong>, <strong>SAS</strong>, and <strong>AMR</strong> super-populations with 
different SNP sets are available for download: 
(1) <a href="https://drive.google.com/file/d/1D0ZtPrvkvznKfHjlwo6BkYIvWFIRF_9g/view?usp=drive_link">All HapMap Variants</a>,(2) <a href="https://drive.google.com/file/d/1c3BamSz1JpXr0QQIfyI_1nUfe0SRNXKa/view?usp=drive_link">All 1000G variants with MAF&gt;5% in any of the five 1000G super-populations</a>,(3) <a href="https://drive.google.com/file/d/18xgsyDymbjeSOv1QS2ULQFaru5NVjwGR/view?usp=drive_link">All 1000G variants with MAF&gt;1% in any of the five 1000G super-populations</a>.</p>
<p>To run bridgePRS using a custom LD reference panel see <a href="../guide_customization/">customisation</a>.</p>
<h2 id="config-files">Config Files</h2>
<p>For the target and base populations BridgePRS requires inputs be supplied in population configuration files.  The following arguments are required in the target configuration file: </p>
<table>
<thead>
<tr>
<th style="text-align: left;">Variable Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">POP</td>
<td style="text-align: left;">Population name label, eg. AFR</td>
</tr>
<tr>
<td style="text-align: left;">LDPOP</td>
<td style="text-align: left;">Name of 1000G population used to estimate LD (AFR, AMR, EAS, EAS, SAS)</td>
</tr>
<tr>
<td style="text-align: left;">LD_PATH</td>
<td style="text-align: left;">Path to LD reference data</td>
</tr>
<tr>
<td style="text-align: left;">SUMSTATS_FILE</td>
<td style="text-align: left;">Sumstats filename used for genomewide summary statistics, can be gzipped</td>
</tr>
<tr>
<td style="text-align: left;">SUMSTATS_PREFIX</td>
<td style="text-align: left;">Sumstats prefix used for summary statistics split by chr, eg. <code>height_chr</code></td>
</tr>
<tr>
<td style="text-align: left;">SUMSTATS_SUFFIX</td>
<td style="text-align: left;">sumstats suffix, eg <code>.txt.gz</code>, gives sumstats files <code>height_chr1.txt.gz -- height_chr22.txt.gz</code></td>
</tr>
<tr>
<td style="text-align: left;">SUMSTATS_SIZE</td>
<td style="text-align: left;">GWAS sample size (required for multi-ancestry)</td>
</tr>
<tr>
<td style="text-align: left;">GENOTYPE_PREFIX</td>
<td style="text-align: left;">Path and prefix to plink bed files of test samples</td>
</tr>
<tr>
<td style="text-align: left;">PHENOTYPE_FILE</td>
<td style="text-align: left;">Filename of phenotype(s) and covariates of test data</td>
</tr>
</tbody>
</table>
<p>The following optional inputs can also be supplied: </p>
<table>
<thead>
<tr>
<th style="text-align: left;">Variable Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">VALIDATION_FILE</td>
<td style="text-align: left;">Filename of phenotype(s) and covariates of validation data</td>
</tr>
<tr>
<td style="text-align: left;">SNP_FILE</td>
<td style="text-align: left;">File listing SNP ids to use in analysis, eg. QCed SNPs</td>
</tr>
<tr>
<td style="text-align: left;">MAX_CLUMP_SIZE</td>
<td style="text-align: left;">Maximum number of SNPs per clump</td>
</tr>
<tr>
<td style="text-align: left;">COVARIATES</td>
<td style="text-align: left;">List of covariates (comma separated) to use, eg: COVARIATES=PC1,PC2,PC3</td>
</tr>
</tbody>
</table>
<p>bridgePRS requires GWAS summary statistics with SNP id (to match to LD
reference data), reference allele, alternate allele, p-value, and
effect size (linear regression coefficient or log odds). Columns can
be supplied in any order. Column names are specified in one of two
ways as described below:</p>
<table>
<thead>
<tr>
<th style="text-align: left;">Variable Name</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">SUMSTATS_FIELDS</td>
<td style="text-align: left;">List of summary statistic column names in the order listed below, e.g, <code>SUMSTATS_FIELDS=ID,REF,A1,P,BETA</code></td>
</tr>
<tr>
<td style="text-align: left;">SSF-SNPID</td>
<td style="text-align: left;">Column name of SNP id (default ID)</td>
</tr>
<tr>
<td style="text-align: left;">SSF-REF</td>
<td style="text-align: left;">Column name of non-effect allele (default REF)</td>
</tr>
<tr>
<td style="text-align: left;">SSF-ALT</td>
<td style="text-align: left;">Column name of effect allele (default ALT)</td>
</tr>
<tr>
<td style="text-align: left;">SSF-P</td>
<td style="text-align: left;">Column name of P-value (default P)</td>
</tr>
<tr>
<td style="text-align: left;">SSF-BETA</td>
<td style="text-align: left;">Column name of effect size (default BETA)</td>
</tr>
</tbody>
</table>
<!--
|Input|Variable Name|Example|
|:-:|:-:|:-:|
|Sumstats Fields    |SUMSTATS_FIELDS   | DEFAULT: SUMSTATS_FIELDS=ID,REF,A1,P,BETA| 
|b1) SNP Field     |SSF-SNPID   | Sumstats Field Name for SNP-ID | 
|b2) Field     |SSF-REF    | Sumstats Field Name for Ref Base |
|b3) Alt Field     |SSF-ALT    | Sumstats Field Name for Alt Base |
|b4) Pval Field     |SSF-P    | Sumstats Field Name for Pvalue | 
|b5) Beta Field     |SSF-BETA   | Sumstats Field Name for Beta | 
-->

<hr />
<h2 id="creating-a-config-file">Creating a Config File:</h2>
<div class="admonition tip">
<p class="admonition-title">Creating a Configuration File</p>
<p>The following command will validate the command line data and create a target configuration file 
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>./bridgePRS tools check-pop -o test --pop AFR --ld_path ./data/1000G_sample/ --sumstats_prefix ./data/pop_AFR/sumstats/AFR.chr 
                                                                                 --sumstats_size   10000
                                                                                 --genotype_prefix ./data/pop_AFR/genotypes/chr 
                                                                                 --phenotype_file ./data/pop_AFR/phenotypes/AFR_test.dat
</code></pre></div>
To create a target and base configuation file you can use the following command: 
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>./bridgePRS tools check-pops -o test --pop AFR EUR --ld_path ./data/1000G_sample/ --sumstats_prefix ./data/pop_AFR/sumstats/AFR.chr ./data/pop_EUR/sumstats/EUR.chr 
                                                                                 --sumstats_size   10000 100000 
                                                                                 --genotype_prefix ./data/pop_AFR/genotypes/chr 
                                                                                 --phenotype_file ./data/pop_AFR/phenotypes/AFR_test.dat
</code></pre></div></p>
</div>
<p>This command will create target and base configuration files can be observed below: </p>
<h2 id="config-files-targetbase-population-data">Config Files: Target/Base Population Data:</h2>
<div class="tabbed-set" data-tabs="1:2"><input checked="checked" id="__tabbed_1_1" name="__tabbed_1" type="radio" /><label for="__tabbed_1_1">cat test/save/target.AFR.config</label><div class="tabbed-content">
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>POP<span style="color: #666666">=</span>AFR
LDPOP<span style="color: #666666">=</span>AFR
LD_PATH<span style="color: #666666">=$</span>BRIDGEDIR<span style="color: #666666">/</span>data<span style="color: #666666">/1000</span>G_sample
SUMSTATS_PREFIX<span style="color: #666666">=</span>test<span style="color: #666666">/</span>save<span style="color: #666666">/</span>sumstats<span style="color: #666666">/</span>ss.AFR.
SUMSTATS_SUFFIX<span style="color: #666666">=</span>.out.gz
SUMSTATS_FIELDS<span style="color: #666666">=</span>ID,REF,A1,P,BETA
SUMSTATS_SIZE<span style="color: #666666">=10000</span>
SNP_FILE<span style="color: #666666">=</span>test<span style="color: #666666">/</span>save<span style="color: #666666">/</span>snps.afr_valid.txt
GENOTYPE_PREFIX<span style="color: #666666">=$</span>BRIDGEDIR<span style="color: #666666">/</span>data<span style="color: #666666">/</span>pop_AFR<span style="color: #666666">/</span>genotypes<span style="color: #666666">/</span>chr
PHENOTYPE_FILE<span style="color: #666666">=</span>tests<span style="color: #666666">/</span>save<span style="color: #666666">/</span>AFR.test_phenos.dat
VALIDATION_FILE<span style="color: #666666">=</span>test<span style="color: #666666">/</span>save<span style="color: #666666">/</span>AFR.valid_phenos.dat
</code></pre></div>
</div>
<input id="__tabbed_1_2" name="__tabbed_1" type="radio" /><label for="__tabbed_1_2">cat out/save/base.EUR.config</label><div class="tabbed-content">
<div class="highlight" style="background: #f8f8f8"><pre style="line-height: 125%;"><span></span><code>POP<span style="color: #666666">=</span>EUR
LDPOP<span style="color: #666666">=</span>EUR
LD_PATH<span style="color: #666666">=$</span>BRIDGEDIR<span style="color: #666666">/</span>data<span style="color: #666666">/1000</span>G_sample
SUMSTATS_PREFIX<span style="color: #666666">=</span>test<span style="color: #666666">/</span>save<span style="color: #666666">/</span>sumstats<span style="color: #666666">/</span>ss.EUR.
SUMSTATS_SUFFIX<span style="color: #666666">=</span>.out.gz
SUMSTATS_FIELDS<span style="color: #666666">=</span>ID,REF,A1,P,BETA
SUMSTATS_SIZE<span style="color: #666666">=100000</span>
SNP_FILE<span style="color: #666666">=</span>test<span style="color: #666666">/</span>save<span style="color: #666666">/</span>snps.afr_valid.txt
GENOTYPE_PREFIX<span style="color: #666666">=$</span>BRIDGEDIR<span style="color: #666666">/</span>data<span style="color: #666666">/</span>pop_AFR<span style="color: #666666">/</span>genotypes<span style="color: #666666">/</span>chr
PHENOTYPE_FILE<span style="color: #666666">=$</span>BRIDGEDIR<span style="color: #666666">/</span>data<span style="color: #666666">/</span>pop_AFR<span style="color: #666666">/</span>phenotypes<span style="color: #666666">/</span>AFR_test.dat
</code></pre></div>
</div>
</div>
<hr />
<h2 id="file-specifications">File Specifications</h2>
<h3 id="1-sumstats-data">1) Sumstats Data</h3>
<p>GWAS summary statistics are provided using a prefix to one or many (per chromosome) files with the <code>--sumstats_prefix</code> argument and the 
<code>--sumstats_suffix</code> argument when nevecessary.  GWAS summary statistics must be provided as a whitespace delimited file containing 
the results of an association study for a given phenotype.  BridgePRS has no problem reading in a gzipped base file 
(need to have a <strong>.gz</strong> suffix) or splitting the file by chromosome if necessary.  An example of a sumstats file with default column headers is shown: </p>
<table>
<thead>
<tr>
<th style="text-align: center;">Defaults</th>
<th style="text-align: center;">#CHR</th>
<th style="text-align: center;">ID</th>
<th style="text-align: center;">REF</th>
<th style="text-align: center;">A1</th>
<th style="text-align: center;">P</th>
<th style="text-align: center;">BETA</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">Variable</td>
<td style="text-align: center;"></td>
<td style="text-align: center;">--ssf-snpid</td>
<td style="text-align: center;">--ssf-ref</td>
<td style="text-align: center;">--ssf-alt</td>
<td style="text-align: center;">--ssf-p</td>
<td style="text-align: center;">--ssf-beta</td>
</tr>
<tr>
<td style="text-align: center;">Data</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">rs121</td>
<td style="text-align: center;">T</td>
<td style="text-align: center;">G</td>
<td style="text-align: center;">0.0413692</td>
<td style="text-align: center;">0.9472871</td>
</tr>
<tr>
<td style="text-align: center;">Data</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">rs497</td>
<td style="text-align: center;">C</td>
<td style="text-align: center;">A</td>
<td style="text-align: center;">0.328347</td>
<td style="text-align: center;">-1.193074</td>
</tr>
<tr>
<td style="text-align: center;">Data</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">rs271</td>
<td style="text-align: center;">G</td>
<td style="text-align: center;">G</td>
<td style="text-align: center;">0.0132225</td>
<td style="text-align: center;">0.413687</td>
</tr>
</tbody>
</table>
<h3 id="2-genotype-files">2) Genotype Files</h3>
<p>Genotype files must be in binary plink format (bed, bim, fam).</p>
<h3 id="3-phenotype-files">3) Phenotype Files</h3>
<p>Phenotype files are provided to BridgePRS using
<code>--phenotype_files</code> for test data and <code>--</code>.  This must be a tab / space delimited file
and missing data <strong>must</strong> be represented by either <code>NA</code> or <code>-9</code> (only
for binary traits).  The first two column of the phenotype file should
be the FID and the IID, and the rest can be phenotypes/covariates:</p>
<table>
<thead>
<tr>
<th style="text-align: center;">FID</th>
<th style="text-align: center;">IID</th>
<th style="text-align: center;">y</th>
<th style="text-align: center;">y.binary</th>
<th style="text-align: center;">PC1</th>
<th style="text-align: center;">PC2</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: center;">afr1_1</td>
<td style="text-align: center;">afr2_1</td>
<td style="text-align: center;">24.4</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0.53</td>
<td style="text-align: center;">0.950</td>
</tr>
<tr>
<td style="text-align: center;">afr1_2</td>
<td style="text-align: center;">afr2_2</td>
<td style="text-align: center;">4.10</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0.59</td>
<td style="text-align: center;">0.450</td>
</tr>
<tr>
<td style="text-align: center;">afr1_3</td>
<td style="text-align: center;">afr2_3</td>
<td style="text-align: center;">37.2</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0.73</td>
<td style="text-align: center;">-0.13</td>
</tr>
<tr>
<td style="text-align: center;">afr1_4</td>
<td style="text-align: center;">afr2_4</td>
<td style="text-align: center;">5.40</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0.44</td>
<td style="text-align: center;">-0.55</td>
</tr>
</tbody>
</table>
<p>The phenotype of interest can be specified with the <code>--phenotype</code> flag and the covariates can be given as a comma separated list 
after the <code>--covariates</code> flag: </p>
<h3 id="4-qc-snp-list">4) QC SNP List</h3>
<p>To select only SNPS that have passed QC, you can include a single column text file using the  <code>--snp_file</code> flag. </p>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../guide_commands/" class="btn btn-neutral float-left" title="Commands"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../guide_output/" class="btn btn-neutral float-right" title="Output Data">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/tadesouaiaia/BridgePRS-website" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../guide_commands/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../guide_output/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-MML-AM_CHTML"></script>
      <script src="../javascripts/details.js"></script>
      <script src="../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
