![Screenshot](img/slim/guide_logo6.png) 

This page contains all command available in BridgePRS. 

!!! tips
    
    When constructing new parameters, we follow the following rule: if the command has effect on any file that is not the target, 
    it will have a prefix of the file name. For example, `--base-info` applies INFO score filtering on the base file, `--ld-info` perform INFO score filtering on the LD reference file and `--info` applies the INFO score filtering on the target file. 


## Input Population Files/Arguments


- `--sumstats_prefix`  Path to sumstats data 

- `--sumstats_suffix`
     Sumstats Suffix        

- `--phenotype_files`
     phenotype test and validation data    


- `--pop_config(s)`
     population configuration file(s) 

- `--pop or --pops`
     population names (AFR, EUR, etc) 

- `--snp_file` List of QCed SNP ids 


## System Level Toggles/Arguments 


- `--cores` By default bridgePRS is parralelized across (n-1) cores 

- `-o`  Output folder for BridgePRS 



- `--platform`  Force platform (Linux or MacOS) 


- `--verbose`  Toggle Verbose Mode On   


- `--noPlots`  Skip Post-Pipeline Analysis (Plotting) 

- `--restart`  Repeat previously completed steps


## Parameter Arguments 


- `--fst` fst between populations analysed

- `--strand_check` Logical to check whether summary statistics and reference data
	input files are on the same strand. If true/1 only unambiguous SNPs are
	used in analyses. If false/0 all variants with matching alleles between
	studies are used. With both options effect and reference alleles are
	checked and flipped where necessary. If a matching pair of alleles is not
	found variants are removed (default 0)

- `--max_clump_size`
     Max Size for Clumping      

- `--thinned_snplist`
     Thinned snp list to use for clumps that exceed
     `--max_clump_size`, e.g. HapMap variants

## Internal File Arguments 

- `--model_file`  model result generated from the build-model subprogram 

- `--clump_prefix` prefix for files generated by clump step   

- `--beta_prefix` prefix for files generated by beta step   

- `--predict_prefix`   prefix for files generated by predict step   

- `--result_file`  prs result files 





## File Column Names 


### Phenotype Files 


- `--phenotype`  Phenotype File Field: phenotype 
- `--covariates` Phenotype File Field: comma separated list of covariates      


### Sumstats Files 



- `--ssf-snpid`
     Sumstats field: snp ID       

- `--ssf-ref`
     Sumstats Field: reference allele      

- `--ssf-alt`
     Sumstats Field: alt allele      

- `--ssf-maf`
     Sumstats Field: MAF       

- `--ssf-beta`
     Sumstats Field: linear regression effect or log odds       

- `--ssf-se`
     Sumstats field: standard error      

- `--ssf-p`
     Sumstats field: P-value       

- `--ssf-n`
     Sumstats Field: Sample Size      





